spark/sql-13/src/itest/scala/org/elasticsearch/spark/integration/AbstractScalaEsSparkSQL.scala

```diff
@@ -372,6 +372,33 @@ class AbstractScalaEsScalaSparkSQL(prefix: String, readMetadata: jl.Boolean, pus
     assertEquals(4l, array(2))
   }
 
+  @Test
+  def testArrayValueWithUserSchema {
+    val index = wrapIndex("sparksql-test-array-value-with-schema")
+    val indexAndType = s"$index/data"
+    RestUtils.touch(index)
+
+    // add some data
+    val jsonDoc = """{"array" : [1, 2, 4, 5] }"""
+    sc.makeRDD(Seq(jsonDoc)).saveJsonToEs(indexAndType)
+    RestUtils.refresh(index)
+
+    val newCfg = collection.mutable.Map(cfg.toSeq: _*) += (ES_READ_FIELD_AS_ARRAY_INCLUDE -> "array")
+    val schemaInt = StructType(StructField("array", ArrayType(IntegerType)) :: Nil)
+
+    val df = sqc.read.options(newCfg).format("org.elasticsearch.spark.sql")
+      .schema(schemaInt).load(indexAndType)
```

Comment:
After checking this out and running it I found that the test does not actually pass. It looks like it passed in CI because of an issue in the build process.

I made some changes to the test locally, namely adding a field to the jsonDoc and omitting it from the schema. The connector fails because it cannot determine the spark field type for the data when it reads it. This makes me think that there are some deeper issues that need to be addressed before we can accept this kind of PR, namely around how we parse the data returned from Elasticsearch.